### PR TITLE
Logic natalia

### PR DESCRIPTION
**Pull request by maxnatalia**
I added logic api for movies list and error/loading component, we need to fix loading page (handle header) and tiles. There're not tags and genres in Popular movies

### CODE REVIEW

File: src/features/movies/moviesSaga.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+// import { apiUrl, apiKey } from "../../apiData";
+import { getMovies } from "./api";
 
-function* fetchMoviesHandler() {
+function* fetchMoviesHandler({ payload }) {
     try {
-        const data = yield axios.get(
-            `${apiUrl}movie/popular?api_key=${apiKey}&language=en-US&page=1`
-        );
+        const data = yield call(getMovies, payload);
```
Comment (golosgit) [inline src/features/movies/moviesSaga.js]: I think you can delete payload here. If I see correctly it's not used (at least not now).

**Review by golosgit — APPROVED** [review]
Generally everything looks good. Before merging you can delete all comments.

File: src/features/movies/MoviesList/index.js [hunk 2]
```diff
@@ -15,114 +15,64 @@ import {
...
+            ) : (
+                <>
+                    <Header>Popular movies</Header>
+                    <TilesContainer insideDetails={insideDetails}>
+                        {movies.map(movie =>
+                            <TileMovie
+                                key={movie.id}
+                            >
+                                <ImageWrapper>
+                                    <Image src={`https://image.tmdb.org/t/p/w400${movie.poster_path}`} alt="" />
```
Comment (kraviecdev) [inline src/features/movies/MoviesList/index.js:52]: Here alt is unnecassary bc there is No in the api

File: src/features/movies/MoviesList/index.js [hunk 3]
```diff
@@ -15,114 +15,64 @@ import {
...
+                                </ImageWrapper>
+                                <ContentWrapper>
+                                    <Title>{movie.original_title}</Title>
```
Comment (kraviecdev) [inline src/features/movies/MoviesList/index.js:55]: The should be movie.title - from almost the end of the object

File: src/features/movies/MoviesList/index.js [hunk 4]
```diff
@@ -15,114 +15,64 @@ import {
...
+                                    <StyledDate>
+                                        {movie.release_date.slice(0, 4)}
+                                    </StyledDate>
+                                    <TagsWrapper>
+                                        <Tag>Action</Tag>
```
Comment (kraviecdev) [inline src/features/movies/MoviesList/index.js:60]: I suggest to also map the tags it got id and name so map by id & display name - movie.genres.id - something like that

File: src/rootSaga.js [hunk 5]
```diff
@@ -1,8 +1,8 @@
 import { all } from "redux-saga/effects";
+import { watchFetchPopularMovies } from "./features/movies/moviesSaga";
 
 export default function* rootSaga() {
     yield all([
-        // moviesSaga(),
```
Comment (kraviecdev) [inline src/rootSaga.js:5]: Delete comments

File: src/store.js [hunk 6]
```diff
@@ -1,13 +1,13 @@
 import { configureStore } from "@reduxjs/toolkit";
 import createSagaMiddleware from "@redux-saga/core";
 import rootSaga from "./rootSaga";
+import moviesReducer from "./features/movies/moviesSlice";
 
 const sagaMiddleware = createSagaMiddleware();
 
 const store = configureStore({
     reducer: {
-        // movies: moviesReducer,
```
Comment (kraviecdev) [inline src/store.js:9]: Delete comments

**Review by kraviecdev — APPROVED** [review]
Delete comments, if tags would be mogę complicated merge the branch and let Grzegorz try

File: src/features/movies/moviesSaga.js (hunk 1 above)
Comment (maxnatalia) [reply src/features/movies/moviesSaga.js]: comments deleted

File: src/features/movies/MoviesList/index.js (hunk 2 above)
Comment (maxnatalia) [reply src/features/movies/MoviesList/index.js:52]: comments deleted

File: src/features/movies/MoviesList/index.js (hunk 3 above)
Comment (maxnatalia) [reply src/features/movies/MoviesList/index.js:55]: in the api we have two titles: original and title (US) - we can added only title

File: src/features/movies/MoviesList/index.js (hunk 4 above)
Comment (maxnatalia) [reply src/features/movies/MoviesList/index.js:60]: but we have created also fetchGenres etc.

File: src/rootSaga.js (hunk 5 above)
Comment (maxnatalia) [reply src/rootSaga.js:5]: comments deleted

**Pull request merged**